src/tokenizer.js

```diff
@@ -769,6 +745,51 @@ export default class Tokenizer {
     return r1 << 4 | r2;
   }
 
+  scanUnicode() {
```

Comment:
Method could be shortened. However, I'm recommending  we leave it as is in order to maintain readability of the code.